presentation/src/main/java/com/exallium/mvvmcleanexample/presentation/utils/notifiable/BaseNotifiableObservable.kt

```diff
@@ -0,0 +1,67 @@
+package com.exallium.mvvmcleanexample.presentation.utils.notifiable
+
+import android.databinding.Bindable
+import android.databinding.Observable
+import android.databinding.PropertyChangeRegistry
+
+class BaseNotifiableObservable : NotifiableObservable {
+
+    private var observable: Observable? = null
+
+    private var initialized = false
+    private val propertyChangeRegistry: PropertyChangeRegistry by lazy {
+        initialized = true
+        PropertyChangeRegistry()
+    }
+
+    constructor() {
+        observable = this
+    }
+
+    constructor(observable: Observable) {
+        this.observable = observable
+    }
+
+    @Synchronized
+    override fun delegateTo(notifiableObservable: NotifiableObservable) {
+        observable = notifiableObservable
+    }
+
+    @Synchronized
+    override fun addOnPropertyChangedCallback(callback: Observable.OnPropertyChangedCallback) {
+        propertyChangeRegistry.add(callback)
+    }
+
+    @Synchronized
+    override fun removeOnPropertyChangedCallback(callback: Observable.OnPropertyChangedCallback) {
+        propertyChangeRegistry.remove(callback)
+    }
+
+    /**
+     * Notifies listeners that all properties of this instance have changed.
+     */
+    @Synchronized
+    override fun notifyChange() {
+        if (initialized) {
```

Comment:
I can remove this class, we probably don't need it. This is merely a work around to not enforce having to inherit from the framework version of BaseObservable and still get all the features from it. This is a re-implementation of it that allows us to delegate the behaviour.

Actually the flag is not needed as long as I hold a reference to the Lazy property delegate (I think)